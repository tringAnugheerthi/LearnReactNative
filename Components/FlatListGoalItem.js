import React from 'react';
import {
    View,
    Text,
    StyleSheet,
    TouchableOpacity
} from 'react-native';

const FlatListGoalItem = props => {

    const onPressHandler = () => props.onDelete(props.item.myOwnKey)

    return (
        <TouchableOpacity onPress={onPressHandler}>
            <View style={styles.listItem}>
                <Text>AG {props.item.value}</Text>
            </View>
        </TouchableOpacity>
    )

}

const styles = StyleSheet.create({
    listItem: {
        padding: 10,
        marginVertical: 10,
        backgroundColor: '#ccc',
        borderColor: 'black',
        borderWidth: 1,
        alignItems: 'center',
        width: '100%',
    }
})

export default FlatListGoalItem;