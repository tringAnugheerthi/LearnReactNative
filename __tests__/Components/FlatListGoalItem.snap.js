import React from 'react';
import { shallow } from 'enzyme';
import FlatListGoalItem from '../../Components/FlatListGoalItem';

const testInputDict = {
    'myOwnKey': Math.random().toString,
    'value': 'Learn react native'
}

describe('Rendering', () => {
    it('Should match snapshot', () => {
        const component = shallow(<FlatListGoalItem item={testInputDict} />);
        expect(component).toMatchSnapshot('Flat List Item Snapshot')
    })
})

// describe('Interaction', () => {
//     describe('On Press', () => {
//         it('Should call on press', () => {
//             const mockOnPress = jest.fn()
//             const component = shallow(<FlatListGoalItem item={testInputDict} />);
//             // const component = shallow(<FlatListGoalItem
//             //     item={testInputDict}
//             //     onPress={mockOnPress} />)

//             const componentInstance = component.instance()
//             console.log(componentInstance)
//             componentInstance.onPressHandler()
//             //../../node_modules/react-native/Libraries/Components/Touchable/
//             // component.find('TouchableOpacity').simulate('click')
//             expect(mockOnPress).toHaveBeenCalled()
//             expect(mockOnPress).toHaveBeenCalledTimes(1)
//         })
//     })
// })