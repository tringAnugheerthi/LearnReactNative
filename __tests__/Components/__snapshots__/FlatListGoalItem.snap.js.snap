// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Should match snapshot: Flat List Item Snapshot 1`] = `ShallowWrapper {}`;
